compiler/rustc_mir_transform/messages.ftl

```diff
@@ -69,8 +69,8 @@ mir_transform_tail_expr_local = {$is_generated_name ->
         *[false] `{$name}` calls a custom destructor
     }
 
-mir_transform_unaligned_packed_ref = reference to packed field is unaligned
-    .note = packed structs are only aligned by one byte, and many modern architectures penalize unaligned field accesses
+mir_transform_unaligned_packed_ref = reference to field of packed {$ty_descr} is unaligned
+    .note = this {$ty_descr} is at most {$align}-byte aligned, but the type of this field may require higher alignment
```

Comment:
Nit: Could you add a test that triggers this diagnostic with a union? Just to ensure that the `$ty_descr` we track is the correct one.